### PR TITLE
[DependencyInjection] Deprecate `!tagged` tag, use `!tagged_iterator` instead

### LINKED ISSUE #47805: [DependencyInjection] !tagged is still working on Symfony DI >= 5.0
### Symfony version(s) affected

5.0.0+

### Description

https://github.com/symfony/symfony/pull/31967 removed support for `!tagged` for Symfony DI 5.0.

That commit got lost along the way somehow (I didn't have time to investigate this in depth).

I've noticed this when doing a code review on our project which uses Symfony 5.4.13, the container was complied without any errors and the functional tests for the service which was using this old syntax were passing so I went digging how is that possible :smile: 

The support for `!tagged` is still there in the Yaml and XML DI file loaders even on the latest Symfony 6.2 branch.

### How to reproduce

```yaml
   Foo\Bar:
        arguments:
            $x: !tagged my_custom_tag
```

I would expect this to blow up during the container compilation on any project that uses Symfony  DI >= 5.0.0, but currently it works fine.

### Possible Solution

Deprecate this again for Symfony 6.2 and remove it in Symfony 7? :smile: 

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 7.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | yes
| Issues        | Fix #47805
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php [hunk 1]
```diff
@@ -586,6 +586,8 @@ private function getArgumentsAsPhp(\DOMElement $node, string $name, string $file
                     $arguments[$key] = new ServiceLocatorArgument($arg);
                     break;
                 case 'tagged':
+                    trigger_deprecation('symfony/dependency-injection', '7.2', 'Type "%s" of tag <%s> in "%s" is deprecated, use "tagged_iterator" or "tagged_locator" instead.', $arg->getAttribute('type'), $name, $file);
```
Comment (stof) [inline src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: the replacement doing the same thing is `tagged_iterator` (see this `$forLocator` variable that is used to handle differences for `tagged_locator`)

**Review by stof — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php (hunk 1 above)
Comment (alexandre-daubois) [reply src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: I was wondering whether indicating both was not a good thing for discovery, but it's actually more accurate to just put the iterator in.

**Comment by alexandre-daubois** [comment]
Status: Needs Review

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php [hunk 2]
```diff
@@ -586,6 +586,8 @@ private function getArgumentsAsPhp(\DOMElement $node, string $name, string $file
                     $arguments[$key] = new ServiceLocatorArgument($arg);
                     break;
                 case 'tagged':
+                    trigger_deprecation('symfony/dependency-injection', '7.2', 'Type "%s" of tag <%s> in "%s" is deprecated, use "tagged_iterator" instead.', $arg->getAttribute('type'), $name, $file);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]:
```suggestion
                    trigger_deprecation('symfony/dependency-injection', '7.2', 'Type "tagged" is deprecated for tag <%s>, use "tagged_iterator" instead in "%s".', $name, $file);
```

File: src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php [hunk 3]
```diff
@@ -852,6 +852,10 @@ private function resolveServices(mixed $value, string $file, bool $isParameter =
                 return new ServiceLocatorArgument($argument);
             }
             if (\in_array($value->getTag(), ['tagged', 'tagged_iterator', 'tagged_locator'], true)) {
+                if ('tagged' === $value->getTag()) {
+                    trigger_deprecation('symfony/dependency-injection', '7.2', 'Using the "!tagged" tag is deprecated, use "!tagged_iterator" instead.');
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php]:
```suggestion
                    trigger_deprecation('symfony/dependency-injection', '7.2', 'Using "!tagged" is deprecated, use "!tagged_iterator" instead in "%s".', $file);
```

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php (hunk 2 above)
Comment (alexandre-daubois) [reply src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: Finding the good wording is always something :smile: I updated both

**Review by stof — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Comment by chalasr** [comment]
Thank you @alexandre-daubois.

**Pull request merged**